source/spec/vagrant-openstack-provider/client/nova_spec.rb

```diff
@@ -212,21 +212,57 @@
         end
       end
 
-      context 'with volume_boot' do
-        it 'returns new instance id' do
+      context 'with volume_boot creating volume' do
+        it 'create bootable volume and returns new instance id' do
           stub_request(:post, 'http://nova/a1b2c3/servers')
             .with(
-              body: '{"server":{"name":"inst","block_device_mapping":[{"volume_id":"vol","device_name":"vda"}],"flavorRef":"flav","key_name":"key"}}',
+              body: '{"server":{"name":"inst","block_device_mapping_v2":[{"boot_index":"0","volume_size":"10","uuid":"image_id",'\
+              '"device_name":"vda","source_type":"image","destination_type":"volume","delete_on_termination":"false"}],'\
+              '"flavorRef":"flav","key_name":"key"}}',
               headers:
-                  {
-                    'Accept' => 'application/json',
-                    'Content-Type' => 'application/json',
-                    'X-Auth-Token' => '123456'
-                  })
+              {
+                'Accept' => 'application/json',
+                'Accept-Encoding' => 'gzip, deflate',
+                'Content-Length' => '248',
+                'Content-Type' => 'application/json',
+                'User-Agent' => 'Ruby',
+                'X-Auth-Token' => '123456'
```

Comment:
Headers `Accept-Encoding`, `Content-Length` and `User-Agent` should be remove. Not useful in unit tests.